install.sh

```diff
@@ -0,0 +1,151 @@
+#! /bin/bash
+
+### Checking for prerequisites ###
+
+#check if running as root
+if [[ $EUID -ne 0 ]]; then
+   echo "This script must be run as root" 
+   exit 1
+fi
+
+#check for docker
+if ! command -v docker &> /dev/null
+then
+  echo "Docker is not installed on this host. Please install and run this script again."
+  exit 1
+fi
+
+#check for docker-compose and install if not currently
+if command -v docker-compose &> /dev/null
+then
+  echo "docker-compose is installed install will continue"
+else
+  read -p "docker-compose not detected as installed. Would you like to install it now? (y/n): " -n 1 -r 
+  echo "/n"
+  if [[ $REPLY =~ ^[yY]$ ]]
+  then
+    curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+    chmod +x /usr/local/bin/docker-compose
+  fi
+fi
+
+### Checking commandline strings ###
+
+if [[ $* == *--[sS][sS][Ll]* ]]
+then
+    echo "Installing in HTTPS mode"
+    SSL="yes"
+else 
+    echo "installing in HTTP mode"
+    SSL="no"
+fi
+
+mkdir ./nginx ./gitlab
+
+if [ $SSL == "yes" ]
```

Comment:
I mentioned this in chat but quoting is tricky. You can drop the quotes here just fine because you're setting the variable and the string is just a literal with no spaces. Safer would be either of these:
```bash
if [ "$SSL" == "yes" ]; then ...
```
or this is safe because it's using double brackets.
```bash
if [[ $SSL == yes ]]; then ...
```